module #2. REST API Basics/rest-api-basic/core/src/main/java/com/epam/esm/repository/impl/jdbctemplate/mysql/GiftCertificateJDBCTemplate.java

```diff
@@ -0,0 +1,91 @@
+package com.epam.esm.repository.impl.jdbctemplate.mysql;
+
+import com.epam.esm.model.GiftCertificate;
+import com.epam.esm.repository.GiftCertificateRepository;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.dao.EmptyResultDataAccessException;
+import org.springframework.jdbc.core.BeanPropertyRowMapper;
+import org.springframework.jdbc.core.JdbcTemplate;
+import org.springframework.jdbc.support.GeneratedKeyHolder;
+import org.springframework.jdbc.support.KeyHolder;
+import org.springframework.stereotype.Repository;
+
+import java.math.BigDecimal;
+import java.sql.PreparedStatement;
+import java.sql.Timestamp;
+import java.util.List;
+import java.util.Objects;
+import java.util.Optional;
+
+@Repository
+public class GiftCertificateJDBCTemplate implements GiftCertificateRepository {
+    private static final String INSERT = "INSERT INTO gift_certificate " +
+            "(name, description, price, duration, create_date) VALUES (?, ?, ?, ?, ?)";
+    private static final String FIND_BY_ID = "SELECT * FROM gift_certificate WHERE id = ?";
+    private static final String FIND_BY_NAME = "SELECT * FROM gift_certificate WHERE name = ?";
+    private static final String FIND_ALL = "SELECT * FROM gift_certificate";
+    private static final String UPDATE = "UPDATE gift_certificate SET name = ?, description = ?,"
+            + "price = ?, duration = ?, last_update_date = ? WHERE id = ?";
+    private static final String DELETE = "DELETE FROM gift_certificate WHERE id = ?";
+
+    private final JdbcTemplate jdbcTemplate;
+
+    public GiftCertificateJDBCTemplate(@Qualifier("prodJDBCTemplate") JdbcTemplate jdbcTemplate) {
+        this.jdbcTemplate = jdbcTemplate;
+    }
+
+    @Override
+    public boolean isExists(GiftCertificate giftCertificate) {
+        return false;
```

Comment:
Looks like isExists method is mocked. And is not used anywhere. Please, remove it from GiftCertificateRepository